File: src/entities/parking_lot/parking_lot.cpp
#include "parking_lot.hpp"

ParkingLot::ParkingLot(){}
ParkingLot::ParkingLot(int id) : id(id) {}
ParkingLot::ParkingLot(int id,  std::string name, std::string street, int num_parking_slots):
                  id(id), name(name), street(street), num_parking_slots(num_parking_slots) {}

void ParkingLot::setId(int id){
    this->id = id;
}

void ParkingLot::setName(std::string name){
    this->name = name;
}

void ParkingLot::setStreet(std::string street){
    this->street = street;
}

void ParkingLot::setNumParkingSlots(int num_parking_slots){
    this->num_parking_slots = num_parking_slots;
}

int ParkingLot::getId() const{
    return id;
}

std::string ParkingLot::getName() const{
    return name;
}

std::string ParkingLot::getStreet() const{
    return street;
}

int ParkingLot::getNumParkingSlots() const{
    return num_parking_slots;
}

ParkingLot& ParkingLot::operator= (const ParkingLot& p1){
    id = p1.id;
    name = p1.name;
    street = p1.street;
    num_parking_slots = p1.num_parking_slots;
    return *this;
}


bool operator== (const ParkingLot& p1, const ParkingLot& p2)
{
    return p1.id == p2.id;
}

std::ostream& operator<<(std::ostream& os, const ParkingLot& p)
{
    os << "ParkingLot - Id: " << p.id << ", Name: " << p.name << ", Street: " << p.street << ", Num parking slots: "
      << p.num_parking_slots << std::endl;
    return os;
}

ParkingLotException::ParkingLotException(const std::string & message):_message(message) {}
const char * ParkingLotException::what() const throw() {
      return _message.c_str();
}

File: uPark_client/entities/parking_lot.py
class ParkingLot:
    def __init__(self, id = None, name = None, street = None, num_parking_slots = None, parking_slots = None):
        self._id = id
        self._name = name
        self._street = street
        self._num_parking_slots = num_parking_slots
        self._parking_slots = parking_slots


    def set_id(self, id):
        self._id = id

    def set_name(self, name):
        self._name = name

    def set_street(self, street):
        self._street = street

    def set_num_parking_slots(self, num_parking_slots):
        self._num_parking_slots = num_parking_slots

    def set_parking_slots(self, parking_slots):
        self._parking_slots = parking_slots


    def get_id(self):
        return self._id

    def get_name(self):
        return self._name

    def get_street(self):
        return self._street

    def get_num_parking_slots(self):
        return self._num_parking_slots

    def get_parking_slots(self):
        return self._parking_slots

    def __eq__(self, other):
        if isinstance(other, ParkingLot):
            return self._id == other._id
        return NotImplemented

    def __str__(self):
        parking_lot_info = (f"Parking lot Id: {self._id}, name: {self._name}, street: {self._street}, num parking slots: {self._num_parking_slots}")
        return parking_lot_info
